pyccel/parser/syntax/headers.py

```diff
@@ -157,6 +157,39 @@ def expr(self):
             d_var['order'] = order
         return d_var
 
+class ShapedID(BasicStmt):
+    """class representing a ShapedID in the grammar.
+
+    Parameters
+    ----------
+    name: str
+        Name of the variable result
+
+    shape: list
+        A list representing the shape of the result
+    """
+
+    def __init__(self, **kwargs):
+        self._name   = kwargs.pop('name')
+        self._shape  = kwargs.pop('shape', [])
+
+        super().__init__(**kwargs)
+
+    @property
+    def expr(self):
+        """Returns a dictionary containing name and shape of result"""
+        d_var = {}
+        shape = []
+        for i in self._shape:
+            if isinstance(i, MacroStmt):
+                shape.append(i.expr)
+            else:
+                shape.append(PyccelSymbol(i))
+        d_var['name']   = self._name
+        d_var['shape']  = shape
```

Comment:
I suspect this would be faster if you used comprehensions:
```
shape = [i.expr if isinstance(i, MacroStmt) else PyccelSymbol(i) for i in self._shape]
d_var = {'name': self._name, 'shape': shape}
```

Reply:
Fixed b70d8ca